libvast/src/schema.cpp

```diff
@@ -219,23 +220,34 @@ caf::expected<schema> get_schema(const caf::settings& options) {
 
 detail::stable_set<std::filesystem::path>
 get_schema_dirs(const caf::actor_system_config& cfg,
-                std::vector<const void*> objpath_addresses) {
+                const std::vector<const void*>& objpath_addresses) {
   const auto disable_default_config_dirs
     = caf::get_or(cfg, "vast.disable-default-config-dirs", false);
   detail::stable_set<std::filesystem::path> result;
   if (auto vast_schema_directories = detail::locked_getenv("VAST_SCHEMA_DIRS"))
     for (auto&& path : detail::split(*vast_schema_directories, ":"))
       result.insert({path});
+  auto insert_dirs_for_datadir = [&](const std::filesystem::path& datadir) {
+    result.insert(datadir / "schema");
+    for (const auto& plugin : plugins::get()) {
+      std::error_code err{};
+      auto dir = datadir / "plugin" / plugin->name() / "schema";
+      if (auto err = std::error_code{}; std::filesystem::exists(dir, err))
+        result.insert(std::move(dir));
+    }
+  };
 #if !VAST_ENABLE_RELOCATABLE_INSTALLATIONS
-  result.insert(VAST_SCHEMADIR);
+  insert_dirs_for_datadir(VAST_DATADIR);
 #endif
```

Comment:
Not the scope of this PR, but shouldn't the loop below be in an `#else` block?

Reply:
That'd break integration tests from within the build directory.